tests/test_parser.py

```diff
@@ -270,6 +341,55 @@ def test_parse_region_fail(text: str):
         parser.parse_optional_region()
 
 
+@pytest.mark.parametrize(
+    "text",
+    [
+        """%x : i32 { "test.op"(%x) : (i32) -> () }""",
+        """%x : i32 { "test.op"(%x) : (i32) -> () ^bb0: }""",
+    ],
+)
+def test_parse_region_with_args(text: str):
+    """Parse a region with args already provided."""
+    ctx = MLContext()
+    ctx.register_dialect(Builtin)
+    ctx.register_dialect(Test)
+
+    parser = Parser(ctx, text)
+    arg = parser.parse_argument()
+    region = parser.parse_region((arg,))
+    assert len(region.blocks[0].args) == 1
+
+    parser = Parser(ctx, text)
+    arg = parser.parse_argument()
+    parser.parse_optional_region((arg,))
```

Comment:
```suggestion
    region = parser.parse_optional_region((arg,))
```
? maybe I'm misunderstanding